packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -14,7 +14,8 @@ import 'tabs.dart';
 import 'theme.dart';
 import 'typography.dart';
 
-final Object _kDefaultHeroTag = new Object();
+const Object _kDefaultHeroTag = const Object();
```

Comment:
We can't use `const Object()` as the default hero tag because const objects are de-duplicated across the system.  That means another file that makes a `const Object()` hero tag will match this one.